### PR TITLE
📖  Use consistent naming for minikube, kind, kOps and Kubespray in documentation

### PR DESCRIPTION
**Pull request by oscr**
**What this PR does / why we need it**:

A minor documentation update to ensure consistent naming for `minikube`, `kind`, `kOps`, and `Kubespray`.

Quick links for spelling reference:
https://kind.sigs.k8s.io/
https://minikube.sigs.k8s.io/docs/
https://github.com/kubernetes/kops
https://kubespray.io/#/docs/comparisons

### CODE REVIEW

**Review by killianmuldoon — APPROVED** [review]
/lgtm

**Comment by sbueringer** [comment]
Please squash before merge

**Comment by sbueringer** [comment]
Thx, can you please squash the commits? Then we should be ready to merge

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ about: Tell us about a problem you are experiencing
 **Environment:**
 
 - Cluster-api version:
-- Minikube/KIND version:
+- Minikube/kind version:
```
[the remark this answers is not in the document]
Comment (oscr) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Thanks for the suggestion @sbueringer!

**Comment by sbueringer** [comment]
Thank you!

/lgtm
/approve

**Pull request merged**